### ISSUE TITLE
making initial commit(s) as part of the scaffolding process

### ISSUE THREAD

**Issue by travi** [form8ion/rfcs#5]
i'd like to handle the workflow for creating new packages (starting with publishing from an alpha branch and avoiding publishes from master until desired) a bit better as part of the scaffolding process. currently this is unique to the js language scaffolder, but assuming use of semantic-release for publishing for other language scaffolders, it is probably pretty safe to assume a very similar workflow, regardless of the particular language scaffolder.

now that we report next-steps after scaffolding a project both to the command line and to github as issues, i think it is ok to make an initial commit that does not account for doing those things first. however, that process should probably be a little different depending on whether the project will be published as a package or not.

* if not published as a package, scaffold all things that are automated and commit the full result to the default branch. then set up the default branch to track the default branch on the vcs host and push.
* if published as a package, scaffold all things _except those that could trigger a release_ and commit that potentially partial result on the default branch. set up the default branch to track the default branch on the vcs host and push. create an `alpha` branch locally, scaffold additional details that are related to publishing, either from the branch or once the branch is merged back into the default branch. commit remaining changes to the `alpha` branch, set it up to track the remote `alpha` branch, and push to the vcs host

**Comment by travi** [comment]
things that could be deferred until the `alpha` branch is created:
* running semantic-release on ci. this is safe as long as no commits are added, but it could be even safer if not configured at all on default until we actually want versions to be published there
* configs that enable dependency update services. if semantic-release is already enabled on the default branch, we dont want dependency updates to be automerged there that would trigger an update. we do want the service to be automatically configured once the branch is merged back to the default branch. for some services, like renovate, we probably want it to be enabled for the branch, even before merging back to the default branch

**Comment by travi** [comment]
## Note:

this two-step process further highlights the need for the ability to generate simple initial versions of some files and the ability to update them with changes later. this is already core to use of `lift`, but is also showing value in some initial scaffolding of files, like the project root `README.md`. 

all of the js ast manipulation tools i've looked into seem to require initial content that is already js so that the tool is focused on modification rather than creation.

more and more, it feels like scaffolding should usually be designed with a two-step process in mind of initial creation and later modification. done well, this has the potential to greatly simplify initial file creation since complex logic can instead be deferred to an update step, even if that step is applied immediately after the creation step.

**Comment by travi** [comment]
capturing some high-level thoughts from our chat tonight:

> for example, i feel like this could be the steps for a new npm package project:
> * scaffold the very core files, like very basic readme, license file, gitattributes, gitignore (possibly empty at this point)
> * commit on default branch. push to vcs host (assuming it will be hosted)
> * create alpha branch and switch to it
> * create additional files like package.json, renovate config
> * commit. push to vcs host
> * finish more complex scaffolding logic. pass results to lift to modify existing files
commit, push
> 
> the alpha branch pieces are pretty specific to packages, so would need to do that piece conditionally, but i highlight that flow because it is probably the more complex option, but also the thing i scaffold far more than apps
